spec/models/in_person_enrollment_spec.rb

```diff
@@ -180,6 +180,115 @@
     end
   end
 
+  describe 'status checks for ready and waiting enrollments' do
+    let(:check_interval) { ...1.hour.ago }
+    let!(:passed_enrollment) do
+      create(:in_person_enrollment, :passed, ready_for_status_check: true)
+    end
+    let!(:failing_enrollment) do
+      create(:in_person_enrollment, :failed, ready_for_status_check: true)
+    end
+    let!(:expired_enrollment) do
+      create(:in_person_enrollment, :expired, ready_for_status_check: true)
+    end
+    let!(:checked_pending_enrollment) do
+      create(
+        :in_person_enrollment, :pending, status_check_attempted_at: Time.zone.now,
+                                         ready_for_status_check: true
+      )
+    end
+    let!(:ready_enrollments) do
+      [
+        create(:in_person_enrollment, :pending, ready_for_status_check: true),
+        create(:in_person_enrollment, :pending, ready_for_status_check: true),
+        create(:in_person_enrollment, :pending, ready_for_status_check: true),
+        create(:in_person_enrollment, :pending, ready_for_status_check: true),
+      ]
+    end
+    let!(:needy_enrollments) do
+      [
+        create(:in_person_enrollment, :pending, ready_for_status_check: false),
+        create(:in_person_enrollment, :pending, ready_for_status_check: false),
+        create(:in_person_enrollment, :pending, ready_for_status_check: false),
+        create(:in_person_enrollment, :pending, ready_for_status_check: false),
+      ]
+    end
+
+    it 'returns only pending enrollments that are ready for status check' do
+      expect(InPersonEnrollment.count).to eq(12)
+      ready_results = InPersonEnrollment.needs_status_check_on_ready_enrollments(check_interval)
+      expect(ready_results.length).to eq ready_enrollments.length
+      expect(ready_results.pluck(:id)).to match_array ready_enrollments.pluck(:id)
+      expect(ready_results.pluck(:id)).not_to match_array needy_enrollments.pluck(:id)
+      ready_results.each do |result|
+        expect(result.pending?).to be_truthy
+      end
+    end
+
+    it 'indicates whether a ready enrollment needs a status check' do
+      expect(passed_enrollment.needs_status_check_on_ready_enrollment?(check_interval)).to(
+        be_falsey,
+      )
+      expect(failing_enrollment.needs_status_check_on_ready_enrollment?(check_interval)).to(
+        be_falsey,
+      )
+      expect(expired_enrollment.needs_status_check_on_ready_enrollment?(check_interval)).to(
+        be_falsey,
+      )
+      expect(checked_pending_enrollment.needs_status_check_on_ready_enrollment?(check_interval)).to(
+        be_falsey,
+      )
+      needy_enrollments.each do |enrollment|
+        expect(enrollment.needs_status_check_on_ready_enrollment?(check_interval)).to(
+          be_falsey,
+        )
+      end
+      ready_enrollments.each do |enrollment|
+        expect(enrollment.needs_status_check_on_ready_enrollment?(check_interval)).to(
+          be_truthy,
+        )
+      end
```

Comment:
You can be stricter here with `true`/`false specifically.

```suggestion
      expect(passed_enrollment.needs_status_check_on_ready_enrollment?(check_interval)).to(
        be(false),
      )
      expect(failing_enrollment.needs_status_check_on_ready_enrollment?(check_interval)).to(
        be(false),
      )
      expect(expired_enrollment.needs_status_check_on_ready_enrollment?(check_interval)).to(
        be(false),
      )
      expect(checked_pending_enrollment.needs_status_check_on_ready_enrollment?(check_interval)).to(
        be(false),
      )
      needy_enrollments.each do |enrollment|
        expect(enrollment.needs_status_check_on_ready_enrollment?(check_interval)).to(
          be(false),
        )
      end
      ready_enrollments.each do |enrollment|
        expect(enrollment.needs_status_check_on_ready_enrollment?(check_interval)).to(
          be(true),
        )
      end
```